code/game/machinery/_machinery.dm

```diff
@@ -953,9 +953,6 @@
 	if(!istype(replacer_tool))
 		return FALSE
 
```

Comment:
```suggestion
	if(!user.Adjacent(src) && !replacer_tool.works_from_distance)
		return FALSE
```

Reply:
keep the ranged check; but make it only matter if we are ranged

Reply:
Are you sure? I believe this is handled on the RPED side already
https://github.com/tgstation/tgstation/blob/9d135c67fd4a1e6bbfb22cb0da8a3bf49516f16d/code/modules/research/stock_parts.dm#L42-L43

Reply:
This _has_ brought to my attention that vending machines currently seem to have a similar issue going on, I'll look into it tomorrow

Reply:
yeaaaaaah seems like they got the same treatment
now:
https://github.com/tgstation/tgstation/blob/9d135c67fd4a1e6bbfb22cb0da8a3bf49516f16d/code/modules/vending/_vending.dm#L1145-L1146
before:
https://github.com/tgstation/tgstation/blob/c1f11f26cef91e7279bd4ac936274f78151ad9b0/code/modules/vending/_vending.dm#L1145-L1146

Reply:
Fix vending machines too since it's a simple one liner

Reply:
yeah gimme a bit, had some Poorly Timed Responsibilities